tests/structure/test_grove_view.py

```
@@ -134,6 +161,46 @@ def test_view_reads_edge_metadata(tmp_path):
+
+    paired = {n.value.start: e for n, e in zip(view.get_in_neighbors(dst), view.get_in_edges(dst))}
+    assert paired == {100: {"w": 1}, 300: {"w": 10}}
+
+    strong = view.get_in_neighbors_if(dst, lambda meta: meta["w"] >= 10)
+    assert [k.value.start for k in strong] == [300]
+
+    paired_list = {k.value.start: meta for k, meta in view.get_in_edge_list(dst)}
+    assert paired_list == {100: {"w": 1}, 300: {"w": 10}}
+
+    # empty for a target with no incoming edges; TypeError/ValueError for None
+    leaf = list(view.intersect(_coord(pg, 100, 200), "chr1"))[0]
```

Ruff (0.16.1)

[warning] Prefer `next(iter(view.intersect(_coord(pg, 100, 200), "chr1")))` over single element slice

Replace with `next(iter(view.intersect(_coord(pg, 100, 200), "chr1")))`

(RUF015)